src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1105,6 +1115,26 @@ private static FunctionResult LoadContentResultBuilder(ObjectParser objectParser
             }
             return new(ErrorType.Create(errorDiagnostic));
         }
+
+          private static FunctionResult ReadEnvironmentVariableResultBuilder (IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, FunctionCallSyntaxBase functionCall, ImmutableArray<TypeSymbol> argumentTypes)
+         {
+            var arguments = functionCall.Arguments.ToImmutableArray();
+           
+            if (argumentTypes.Length != 1 || argumentTypes[0] is not StringLiteralType stringLiteral)
+            {
+                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).CompileTimeConstantRequired()));
+            }
+            var envVariableName = stringLiteral.RawStringValue;
+            var envVariableValue = Environment.GetEnvironmentVariable(envVariableName);
+            
+            //error to fail the build-param with clear message of the missing env var name
+            if (envVariableValue == null)
+            {
+                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).FailedToEvaluateParameter(envVariableName, "Environment variable does not exist")));
+            }
+            return new(ConvertJsonToBicepType(envVariableValue), ConvertJsonToExpression(envVariableValue));
```

Comment:
Worth being precise here:

```suggestion
            return new(
                TypeFactory.CreateStringLiteralType(envVariableValue),
                new StringLiteralExpression(null, envVariableValue))
```